### PR TITLE
Issue 2: Save Image Link

### PR DESCRIPTION
**Pull request by chen-daniel**

### CODE REVIEW

**Comment by chen-daniel** [comment]
[Issue 2: Save Image Link](https://trello.com/c/U1B92G4O/37-issue-2-save-image-link)

File: app/models/image.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Image < ApplicationRecord
+  validates :url, presence: true
+  validate :image_url?
+
+  def image_url?
+    parsed_url = URI.parse(url)
+    errors.add(:url, 'Invalid URL') unless parsed_url.is_a?(URI::HTTP) || parsed_url.is_a?(URI::HTTPS)
```
Comment (raj-al-ghul) [inline app/models/image.rb]: could be worth adding additional cases to the tests to handle all of the cases in the `image_url` validation

File: app/views/application/home.html.erb [hunk 2]
```diff
@@ -1 +1,2 @@
-<h1>Hello World</h1>
\ No newline at end of file
+<h1>Hello World</h1>
+<a href="/images/new">Upload Image</a>
```
Comment (raj-al-ghul) [inline app/views/application/home.html.erb]: should use a named route helper here (like you do in controller/tests)

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class ImagesController < ApplicationController
+  def index
+    @images = Image.all
```
Comment (raj-al-ghul) [inline app/controllers/images_controller.rb]: i thought we were skipping this?

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  def new
+    @image = Image.new
+  end
+
+  def create
+    @image = Image.new(image_params)
+
+    if !@image.save
+      render 'new'
```
Comment (raj-al-ghul) [inline app/controllers/images_controller.rb]: can we test this case?
also lets sync up on the story error task, i dont think that is covered as of now

**Comment by raj-al-ghul** [comment]
great start 🎉 

left some comments, lets go through the trello card together.
also could be good to break out the last commit a little

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  test 'should reject invalid image url and rerender' do
+    post images_path, params: { image: { url: 'asdf' } }
+    assert_response :unprocessable_entity
+  end
+
+  test 'should show an image in the db' do
+    image = Image.create!(url: 'https://www.w3schools.com/w3css/img_lights.jpg')
+    get image_path(image.id)
+    assert_response :ok
+    p response.body
```
Comment (ztmeyer) [inline test/controllers/images_controller_test.rb]: Could we remove this debug statement?

File: .rubocop.yml [hunk 6]
```diff
@@ -6,6 +6,7 @@ AllCops:
     - 'vendor/**/*'
     - 'node_modules/**/*'
     - 'bin/*'
+    - 'config/initializers/*'
```
Comment (ztmeyer) [inline .rubocop.yml]: This could have been a separate commit, but don't worry about changing it 👍

**Pull request merged**